shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/service/CreateStatisticsService.java

```diff
@@ -0,0 +1,120 @@
+package org.shanoir.ng.dataset.service;
+
+import org.shanoir.ng.shared.event.ShanoirEvent;
+import org.shanoir.ng.shared.event.ShanoirEventService;
+import org.shanoir.ng.shared.exception.ErrorModel;
+import org.shanoir.ng.shared.exception.RestServiceException;
+import org.shanoir.ng.utils.DatasetFileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.io.*;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+@Service
+public class CreateStatisticsService {
+    @Autowired
+    ShanoirEventService eventService;
+    @Autowired
+    private DatasetService datasetService;
+    private static final String ZIP = ".zip";
+    private static final String JAVA_IO_TMPDIR = "java.io.tmpdir";
+    private static final Logger LOG = LoggerFactory.getLogger(CreateStatisticsService.class);
+
+    private File recreateFile(final String fileName) throws IOException {
+        File file = new File(fileName);
+        if(file.exists()) {
+            file.delete();
+        }
+        file.createNewFile();
+        return file;
+    }
+
+    @Async
+    @Transactional
+    public void createStats(String studyNameInRegExp, String studyNameOutRegExp, String subjectNameInRegExp, String subjectNameOutRegExp, ShanoirEvent event, String params) throws RestServiceException, IOException {
+        float progress = 0;
+        String tmpDir = System.getProperty(JAVA_IO_TMPDIR);
+        File userDir = DatasetFileUtils.getUserImportDir(tmpDir);
+        File statisticsFile = recreateFile(userDir + File.separator + "shanoirExportStatistics.tsv");
```

Comment:
This means that if the user makes two requests (even with different parameters) in parallel, the file will be overwritten no?